packages/es-components/src/components/controls/range-selector/RangeSelector.js

```diff
@@ -0,0 +1,180 @@
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import { noop } from 'lodash';
+
+const RangeSelectorBase = styled.div`
+  width: 100%;
+`;
+const RangeSelectorSlider = styled.div`
+  height: 5px;
+  border-radius: 5px;
+  background: ${props => props.sliderColor};
+  position: relative;
+`;
+const RangeSelectorProgress = styled.div`
+  height: 5px;
+  left: ${props => `${props.minPercentage}%`};
+  right: ${props => `${100 - props.maxPercentage}%`};
+  border-radius: 5px;
+  background: ${props => props.progressColor};
+  position: absolute;
+`;
+const RangeSelectorInputContainer = styled.div`
+  position: relative;
+`;
+
+const RangeSelectorInput = styled.input`
+  position: absolute;
+  top: -5px;
+  height: 5px;
+  width: 100%;
+  background: none;
+  pointer-events: none;
+  -webkit-appearance: none;
+  &:focus {
+    outline: none;
+  }
+  &::-webkit-slider-thumb {
+    height: 17px;
+    width: 17px;
+    border-radius: 50%;
+    pointer-events: auto;
+    -webkit-appearance: none;
+    background: ${props => props.thumbColor};
+  }
+  &::-moz-range-thumb {
+    height: 17px;
+    width: 17px;
+    border: none;
+    pointer-events: auto;
+    -moz-appearance: none;
+    background: ${props => props.thumbColor};
+  }
+`;
+
+function RangeSelector({
+  currentMinValue,
+  currentMaxValue,
+  minValue,
+  maxValue,
+  progressColor,
+  sliderColor,
+  thumbColor,
+  onChange,
+  ...RangeSelectorProps
+}) {
+  const initStateMin = () => {
+    if (currentMinValue < minValue) return 0;
+    if (currentMinValue > maxValue) return 100;
+    return ((currentMinValue - minValue) / (maxValue - minValue)) * 100;
+  };
+  const initStateMax = () => {
+    if (currentMaxValue > maxValue) return 100;
+    if (currentMaxValue < minValue) return 0;
+    return ((currentMaxValue - minValue) / (maxValue - minValue)) * 100;
+  };
+  const [min, setMin] = useState(currentMinValue);
+  const [max, setMax] = useState(currentMaxValue);
+  const [minPercentage, setMinPercentage] = useState(initStateMin);
+  const [maxPercentage, setMaxPercentage] = useState(initStateMax);
```

Comment:
Since we're not doing anything other than rendering them out, these percentages don't need to be in state. They can just be derived inline from `min` and `max`.

Reply:
removed minPercentage  maxPercentage state, make it derived.